File: sdks/thunkmetrc-py/wrapper/src/thunkmetrc/wrapper/services/facilities_service.py
from typing import Any, Optional, List, AsyncGenerator
from thunkmetrc.client import MetrcClient
from ..ratelimiter import MetrcRateLimiter
from ..models import *
from ..utils import iterate_pages, deserialize_page

class FacilitiesService:
    def __init__(self, client: MetrcClient, limiter: MetrcRateLimiter):
        self.client = client
        self.limiter = limiter

    async def get_facilities(self, body: Any = None) -> List['Facility']:
        """
        This endpoint provides a list of facilities for which the authenticated user has access.
        GET /facilities/v2
        Parameters:
        """
        async def op():
            return await self.client.facilities.get_facilities(body=body,
            )
        return await self.limiter.execute(None, True, op)

File: sdks/thunkmetrc-ts/wrapper/src/services/FacilitiesService.ts

import { MetrcClient as InternalMetrcClient } from '@thunkier/thunkmetrc-client';
import { MetrcRateLimiter } from '../RateLimiter';
import { iteratePages } from '../Utils';
import { 
    PaginatedResponse,
    Facility,
} from '../models';

export class FacilitiesService {
  constructor(private client: InternalMetrcClient, private rateLimiter: MetrcRateLimiter) {}

  /**
   * This endpoint provides a list of facilities for which the authenticated user has access.
   * GET GetFacilities
   */
  public async getFacilities(body?: any): Promise<Facility[]> {
    return this.rateLimiter.execute(null,true,
        () => this.client.getFacilities(body,
        )
    );
  }
}

